R/Module-CohortGenerator.R

```diff
@@ -119,25 +119,37 @@ CohortGeneratorModule <- R6::R6Class(
       if (!CohortGenerator::isCohortDefinitionSet(cohortDefinitionSet)) {
         stop("cohortDefinitionSet is not properly defined")
       }
+      sharedResource <- list()
+      templateDefinitions <- CohortGenerator::getTemplateDefinitions(cohortDefinitionSet)
+      if (length(templateDefinitions) > 0) {
+        # Don't save templates as regular cohorts, this is handled entirely in the template definitions
+        cohortDefinitionSet <- cohortDefinitionSet |>
+          dplyr::filter(!.data$isTemplatedCohort)
+
+        templateDefs <- lapply(templateDefinitions, function(x) { x$toList() })
+        names(templateDefs) <- NULL
+        sharedResource$templateDefs <- templateDefs
+      }
 
       subsetDefinitions <- CohortGenerator::getSubsetDefinitions(cohortDefinitionSet)
+
       if (length(subsetDefinitions) > 0) {
         # Filter the cohort definition set to the "parent" cohorts.
         parentCohortDefinitionSet <- cohortDefinitionSet[!cohortDefinitionSet$isSubset, ]
       } else {
         parentCohortDefinitionSet <- cohortDefinitionSet
       }
 
-      sharedResource <- list()
+
       cohortDefinitionSetFiltered <- private$.listafy(parentCohortDefinitionSet)
-      sharedResource["cohortDefinitions"] <- list(cohortDefinitionSetFiltered)
+      sharedResource$cohortDefinitions <- cohortDefinitionSetFiltered
```

Comment:
note, the use of single brackets here is why this needed to be wrapped in a list. Using the `$` notation is the more common convetion, `listVar["varName"]` is pythonic